rpc/json/ws.go

```diff
@@ -40,6 +40,27 @@ func (h *handler) wsHandler(w http.ResponseWriter, r *http.Request) {
 		WriteBufferSize: 1024,
 	}
 
+	// TODO: use CORSAllowedMethods, CORSAllowedHeaders
+	if h.config.IsCorsEnabled() {
+		upgrader.CheckOrigin = func(r *http.Request) bool {
+			origin := r.Header.Get("Origin")
+			if origin == "" {
+				return true
+			}
+
+			allowedOrigins := h.config.CORSAllowedOrigins
+
+			if len(allowedOrigins) == 1 && allowedOrigins[0] == "*" {
```

Comment:
why do you need to check the individual values like `""` and `*`? can you not just do:
```
if h.config.IsCorsEnabled() {
	upgrader.CheckOrigin = func(r *http.Request) bool {
		for _, allowedOrigin := range h.config.CORSAllowedOrigins {
			if allowedOrigin == r.Header["Origin"][0] {
				return true
			}
		}
		return false
	}
}
```

Reply:
Yep, we can do this, but just need it to handle case where origin is not set, so that this doesn't throw index out of range.

Reply:
Commented above on @tzdybal review about how strict or permissive policy we should decide.